src/rpa_gw.F

```diff
@@ -3621,6 +3558,43 @@ SUBROUTINE continuation_pade(vec_gw_energ, vec_omega_fit_gw, &
          CPABORT("Only NONE, Z_SHOT, NEWTON, and BISECTION crossing search implemented.")
       END SELECT
 
+      IF (print_self_energy) THEN
+
+         IF (n_level_gw < 10) THEN
+            WRITE (filename, "(A21,I1)") "self_energy_level_000", n_level_gw_ref
+         ELSE IF (n_level_gw < 100) THEN
+            WRITE (filename, "(A20,I2)") "self_energy_level_00", n_level_gw_ref
+         ELSE IF (n_level_gw < 1000) THEN
+            WRITE (filename, "(A19,I3)") "self_energy_level_0", n_level_gw_ref
+         ELSE
+            WRITE (filename, "(A18,I4)") "self_energy_level_", n_level_gw_ref
+         END IF
+
+         OPEN (UNIT=n_level_gw_ref*1000, FILE=TRIM(filename))
```

Comment:
You can remove the warning from the convention checker by using the routines open_file and close_file from the module cp_files. Further, why do you have different units when you are opening and closing exactly one file in the routine?

Reply:
Thanks a lot for your hint! Will be changed very soon. 

The routine prints the self-energy Sigma(w) for every level that is corrected by GW, so several files are printed. Each level is unique to a single MPI rank and this MPI rank prints the specific file. Does this answer your question?